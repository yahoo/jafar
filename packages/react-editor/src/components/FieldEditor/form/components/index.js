/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import TextInput from '@jafar/react-components/edit/Text';
import Checkbox from '@jafar/react-components/edit/Checkbox';
import MultiSelectFields from './MultiSelectFields';
import Validators from './Validators';
import Handler from './Handler';
import Term from './Term';

export default {
  TextInput,
  Checkbox,
  MultiSelectFields,
  Validators,
  Handler,
  Term,
};
