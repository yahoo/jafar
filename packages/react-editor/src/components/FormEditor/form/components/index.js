/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import TextInput from '@jafar/react-components/edit/Text';
import NumberInput from '@jafar/react-components/edit/Number';
import JsonEditor from '@jafar/react-components/edit/JsonEditor';
import Fields from './Fields';

export default {
  TextInput,
  NumberInput,
  JsonEditor,
  Fields,
};
