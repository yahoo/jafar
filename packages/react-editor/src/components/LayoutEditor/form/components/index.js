/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import TextInput from '@jafar/react-components/edit/Text';
import NumberInput from '@jafar/react-components/edit/Number';
import Select from '@jafar/react-components/edit/Select';
import DynamicItemsModal from '@jafar/react-components/edit/DynamicItemsModal';

export default {
  TextInput,
  NumberInput,
  Select,
  DynamicItemsModal,
};
