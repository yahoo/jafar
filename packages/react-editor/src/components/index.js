/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import FieldEditor from './FieldEditor';
import FormEditor from './FormEditor';

export {
  FieldEditor,
  FormEditor,
};

