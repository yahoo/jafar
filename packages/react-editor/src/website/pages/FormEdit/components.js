/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import _TextInput from '@jafar/react-components/edit/Text';
import _NumberInput from '@jafar/react-components/edit/Number';
import _Checkbox from '@jafar/react-components/edit/Checkbox';
import _CheckboxCollection from '@jafar/react-components/edit/CheckboxCollection';
import _Select from '@jafar/react-components/edit/Select';
import _MultiSelect from '@jafar/react-components/edit/MultiSelect';
import _JsonEditor from '@jafar/react-components/edit/JsonEditor';
import _Switch from '@jafar/react-components/edit/Switch';
import _Url from '@jafar/react-components/view/Url';
import _Boolean from '@jafar/react-components/view/Boolean';
import _JsonView from '@jafar/react-components/view/JsonView';

export const TextInput = _TextInput;
export const NumberInput = _NumberInput;
export const Checkbox = _Checkbox;
export const CheckboxCollection = _CheckboxCollection;
export const Select = _Select;
export const MultiSelect = _MultiSelect;
export const JsonEditor = _JsonEditor;
export const Switch = _Switch;
export const Url = _Url;
export const JsonView = _JsonView;
export const Boolean = _Boolean;

// contains only common components of this app that any entity can use
export default {
  TextInput,
  NumberInput,
  Checkbox,
  CheckboxCollection,
  Select,
  MultiSelect,
  JsonEditor,
  Url,
  Switch,
  JsonView,
  Boolean,
};
