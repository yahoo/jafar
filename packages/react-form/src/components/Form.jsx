/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import React from 'react';
import PropTypes from 'prop-types';
import JForm from '@jafar/form';
import FormContext from './context';

export default class Form extends React.Component {
  static contextType = FormContext;

  static propTypes = {
    model: PropTypes.object.isRequired,
    resources: PropTypes.object,
    settings: PropTypes.object,
    data: PropTypes.object,
    context: PropTypes.object,
  };

  constructor(props) {
    super(props);
    // stored to compare pointer changes for re-init / other actions (react don't pass prevState in getDerivedStateFromProps)
    this.state = {
      propsModel: props.model, 
      propsContext: props.context,
      propsData: props.data,
    };
  }

  static getDerivedStateFromProps(props, state) {
    // if model was changed (checking only for new pointer) - trigger init action
    if (props.model !== state.propsModel) {
      return ({ propsModel: props.model, needAction: 'init', destroyPrevModelId: state.propsModel.id });
    }
    // if context was changed (checking only for new pointer) - trigger changeContext action
    if (props.context !== state.propsContext) {
      return ({ propsContext: props.context, needAction: 'changeContext' });
    }
    // if data was changed (checking only for new pointer) - trigger changeData action
    if (props.data !== state.propsData) {
      return ({ propsData: props.data, needAction: 'changeData' });
    }
    return null;
  }

  render() {
    if (!this.state.form) {
      return null;
    }

    const context = {
      model: this.state.form.model,
      resources: this.state.form.resources,
      actions: {
        changeValue: this.privateForm.changeValue.bind(this.privateForm),
        changeData: this.privateForm.changeData.bind(this.privateForm),
        changeContext: this.privateForm.changeContext.bind(this.privateForm),
        changeState: this.privateForm.changeState.bind(this.privateForm),
        changeUi: this.privateForm.changeUi.bind(this.privateForm),
        submit: this.privateForm.submit.bind(this.privateForm),
        reset: this.privateForm.reset.bind(this.privateForm),
      },
      parent: this.context,
    };

    return (
      <FormContext.Provider value={context}>
        <React.Fragment>{this.props.children}</React.Fragment>
      </FormContext.Provider>);
  }

  componentDidMount() {
    this.init();
  }

  componentDidUpdate() {
    switch (this.state.needAction) {
    case 'init': {
      if (this.state.destroyPrevModelId !== this.state.propsModel.id) {
        this.privateForm.destroy();
      }
      this.init();
      this.setState({ needAction: undefined, destroyPrevModelId: undefined });
      break;
    }
    case 'changeContext': {
      this.privateForm.changeContext(this.props.context);
      this.setState({ needAction: undefined });
      break;
    }
    case 'changeData': {
      this.privateForm.changeData(this.props.data);
      this.setState({ needAction: undefined });
      break;
    }
    default: {}
    }
  }

  init = () => {
    this.privateForm = new JForm();
    const contextOverride = this.props.context ? { context: this.props.context } : undefined;
    const model = Object.assign({}, this.props.model, contextOverride);
    this.privateForm.init(model, this.props.resources, this.props.settings, this.updatePublicForm);
  }

  componentWillUnmount() {
    this.privateForm.destroy();
    this.destroyed = true;
  }

  updatePublicForm = (form) => {
    if (this.destroyed) return;

    this.setState({ form });
  };
}
