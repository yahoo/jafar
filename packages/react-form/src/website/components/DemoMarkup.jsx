/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import React from 'react';
import PrismCode from 'react-prism';
import JSZip from 'jszip';
import 'prismjs';
import 'prismjs/themes/prism.css';
import Styled from './StyledComponents';

export default class DemoMarkup extends React.Component {
  constructor(props) {
    super(props);
    this.formFolder = this.props.formFolder || 'form';
    this.index = this.getFile('index');
    this.fields = this.getFile('fields');
    this.components = this.getFile('components');
    this.data = this.props.data ? this.getFile('data') : undefined;
    this.conversions = this.props.conversions ? this.getFile('conversions') : undefined;
    this.validators = this.props.validators ? this.getFile('validators') : undefined;
    this.terms = this.props.terms ? this.getFile('terms') : undefined;
    this.dependenciesChanges = this.props.dependenciesChanges ? this.getFile('dependencies-changes') : undefined;
    this.hooks = this.props.hooks ? this.getFile('hooks') : undefined;
    this.mockService = this.props.mockService ? this.getFile('mocks/service') : undefined;
    this.extraComponents = this.props.extraComponents ? this.props.extraComponents
      .map(filePath => this.getFileByFullRelativePath(filePath)) : undefined;
  }

  getFile(fileName) {
    return require(`!!raw-loader!../demos/react/${this.props.exampleName}/${this.formFolder}/${fileName}.js`).default;
  }

  getFileByFullRelativePath(filePath) {
    return require(`!!raw-loader!../demos/react/${this.props.exampleName}/${filePath}.jsx`).default;
  }

  download = () => {
    const rootFolder = new JSZip();

    // add form folder
    const formFolder = rootFolder.folder(this.formFolder);
    formFolder.file('index.js', this.index);
    formFolder.file('fields.js', this.fields);
    formFolder.file('components.js', this.components);
    if (this.data) formFolder.file('data.js', this.data);
    if (this.conversions) formFolder.file('conversions.js', this.conversions);
    if (this.validators) formFolder.file('validators.js', this.validators);
    if (this.terms) formFolder.file('terms.js', this.terms);
    if (this.dependenciesChanges) formFolder.file('dependenciesChanges.js', this.dependenciesChanges);
    if (this.hooks) formFolder.file('hooks.js', this.hooks);
    if (this.mockService) {
      const mocksFolder = formFolder.folder('mocks');
      mocksFolder.file('service.js', this.mockService);
    }
    if (this.extraComponents) {
      this.props.extraComponents.forEach((filePath, index) => {
        let file = filePath;
        let containerFolder = rootFolder;
        if (filePath.includes('/')) {
          const arr = filePath.split('/');
          file = arr[1];
          containerFolder = rootFolder.folder(arr[0]);
        }
        containerFolder.file(`${file}.jsx`, this.extraComponents[index]);
      });
    }

    // add demo file
    rootFolder.file('demo.jsx', this.props.demo);

    rootFolder.generateAsync({ type:'blob' }).then((content) => {
      const href = window.URL.createObjectURL(content);
      const link = document.createElement('a');
      link.setAttribute('href', href);
      link.setAttribute('download', `${this.props.exampleName}.zip`);
      link.style.display = 'none';
      document.body.appendChild(link);
      link.click();
      document.body.removeChild(link);
    });
  }

  render() {
    return (<Styled.DemoWrapper>
      <Styled.DownloadDemo onClick={this.download}>Download</Styled.DownloadDemo>
      <Styled.H3>Form definition</Styled.H3>
      <p>{this.formFolder}/fields.js</p>
      <PrismCode className="language-javascript" component="pre">{this.fields}</PrismCode>
      {
        this.data && <div>
          <p>{this.formFolder}/data.js</p>
          <PrismCode className="language-javascript" component="pre">{this.data}</PrismCode>
        </div>
      }
      <p>{this.formFolder}/components.js</p>
      <PrismCode className="language-javascript" component="pre">{this.components}</PrismCode>
      {
        this.props.conversions && <div>
          <p>{this.formFolder}/conversions.js</p>
          <PrismCode className="language-javascript" component="pre">{this.conversions}</PrismCode>
        </div>
      }
      {
        this.props.validators && <div>
          <p>{this.formFolder}/validators.js</p>
          <PrismCode className="language-javascript" component="pre">{this.validators}</PrismCode>
        </div>
      }
      {
        this.props.terms && <div>
          <p>{this.formFolder}/terms.js</p>
          <PrismCode className="language-javascript" component="pre">{this.terms}</PrismCode>
        </div>
      }
      {
        this.props.dependenciesChanges && <div>
          <p>{this.formFolder}/dependencies-changes.js</p>
          <PrismCode className="language-javascript" component="pre">{this.dependenciesChanges}</PrismCode>
        </div>
      }
      {
        this.props.hooks && <div>
          <p>{this.formFolder}/hooks.js</p>
          <PrismCode className="language-javascript" component="pre">{this.hooks}</PrismCode>
        </div>
      }
      <p>{this.formFolder}/index.js</p>
      <PrismCode className="language-javascript" component="pre">{this.index}</PrismCode>
      {
        this.props.mockService && <div>
          <p>{this.formFolder}/mocks/service.js</p>
          <PrismCode className="language-javascript" component="pre">{this.mockService}</PrismCode>
        </div>
      }
      {
        !this.props.hideHtml
        && <div>
          <Styled.H3>Components</Styled.H3>
          <p>Demo.jsx</p>
          <PrismCode className="language-javascript" component="pre">{this.props.demo}</PrismCode>
          {
            this.extraComponents && <div>{
              this.extraComponents.map((componentHtml, index) => (<div key={index}>
                <p>{this.props.extraComponents[index]}.jsx</p>
                <PrismCode className="language-javascript" component="pre">{componentHtml}</PrismCode>
              </div>))
            }</div>
          }
        </div>
      }
    </Styled.DemoWrapper>);
  }
}
