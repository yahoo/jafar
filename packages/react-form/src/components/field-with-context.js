/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import React from 'react';
import PropTypes from 'prop-types';
import { mapFieldToProps } from '@jafar/form';
import FormContext from './context';

const propTypes = {
  id: PropTypes.string.isRequired,
};

export default function withContext(FieldView) {
  class Field extends React.Component {
    static contextType = FormContext;

    static propTypes = propTypes;

    render() {
      const fieldViewProps = mapFieldToProps(this.props.id, this.context.model, this.context.resources);

      return (
        <FieldView
          {...this.props} // custom props to propagate
          {...fieldViewProps}
          onValueChange={this.onValueChange}
          onStateChange={this.onStateChange}
        />
      );
    }

    // We don't pass 'changeValue' and 'changeState' directly to FieldView prop because context 
    // create a new instance of them each render causing un-necessary renders to FieldView
    onValueChange = (id, value) => {
      this.context.actions.changeValue(id, value);
    }

    onStateChange = (id, state) => {
      this.context.actions.changeState(id, state);
    }
  }

  return Field;
}
