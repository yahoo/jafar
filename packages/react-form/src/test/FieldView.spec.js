/**
  * Copyright 2020, Verizon Media
  * Licensed under the terms of the MIT license. See LICENSE file in project root for terms.
  */

import React from 'react';
import PropTypes from 'prop-types';
import { shallow } from 'enzyme';
import { FieldView } from '../components';

describe('FieldView', () => {
  const mockComponent = jest.fn();
  const mockOnValueChange = jest.fn();
  const mockOnStateChange = jest.fn();
  let props;
  let element;

  beforeEach(() => {
    props = {
      id: 'mockId',
      label: 'mockLabel',
      description: 'mockDescription',
      component: mockComponent,
      value: 'mockValue',
      state: {},
      excluded: false,
      disabled: false,
      required: true,
      invalid: false,
      dirty: false,
      empty: false,
      errors: [],
      onValueChange: mockOnValueChange,
      onStateChange: mockOnStateChange,
    };
  });

  describe('Proptypes', () => {
    it('component - valid when component is forwardRef object', () => {
      const component = React.forwardRef((props, ref) => React.createElement('div', { ref, ...props }));
      testComponentPropType(component);
    });

    it('component - valid when component is class component', () => {
      class myInput extends React.Component {
        render() { return (<div>{this.name}</div>); }
      }
      testComponentPropType(myInput);
    });

    it('component - valid when component is functional component', () => {
      const component = () => {};
      testComponentPropType(component);
    });

    function testComponentPropType(component, errorsNumber = 0) {
      const { error } = global.console;
      global.console.error = jest.fn();
      const fieldViewProps = Object.assign(props, { component });
      PropTypes.checkPropTypes(FieldView.propTypes, fieldViewProps, 'prop', 'fieldView');
      expect(global.console.error.mock.calls).toHaveLength(errorsNumber);
      global.console.error = error;
    }
  });

  it('rendered correctly', () => {
    element = shallow(<FieldView {...props} />);
    expect(element).toMatchSnapshot('default values');
  });

  it('not rendered if excluded is true', () => {
    props.excluded = true;
    element = shallow(<FieldView {...props} />);
    expect(element.type()).toEqual(null);
  });

  it('not rendered if no component', () => {
    props.component = undefined;
    element = shallow(<FieldView {...props} />);
    expect(element.type()).toEqual(null);
  });

  it('rendered correctly with invalid and required', () => {
    props.required = true;
    props.errors = [{ name: 'mockErrorMessage' }, { name: 'required' }];
    element = shallow(<FieldView {...props} />);
    expect(element).toMatchSnapshot('invalid field required');
  });

  it('rendered correctly with invalid not required', () => {
    props.invalid = true;
    props.errors = [{ name: 'mockErrorMessage' }];
    element = shallow(<FieldView {...props} />);
    expect(element).toBeTruthy();
    expect(element).toMatchSnapshot('invalid field not required');
  });

  it('rendered correctly with disabled', () => {
    props.disabled = true;
    element = shallow(<FieldView {...props} />);
    expect(element).toBeTruthy();
    expect(element).toMatchSnapshot('with disabled field');
  });

  it('onStateChange calls props.onStateChange', () => {
    element = shallow(<FieldView {...props} />);
    const mockState = { mockState: 'mockState' };
    element.instance().onStateChange(mockState);
    expect(mockOnStateChange).toHaveBeenCalledWith(props.id, mockState);
  });

  it('onValueChange calls props.onValueChange', () => {
    element = shallow(<FieldView {...props} />);
    const mockValue = 'mockValue';
    element.instance().onValueChange(mockValue);
    expect(mockOnValueChange).toHaveBeenCalledWith(props.id, mockValue);
  });

  it('shouldComponentUpdate returns false on when none of the conditions fulfilled', () => {
    element = shallow(<FieldView {...props} />);
    const shouldUpdate = element.instance().shouldComponentUpdate(props);
    expect(shouldUpdate).toBeFalsy();
  });

  it('rendered correctly with unsupported jafar field props - pass it to the underline component', () => {
    element = shallow(<FieldView mickey="mouse" {...props} />);
    expect(element).toMatchSnapshot('with unsupported prop');
  });

  it(`rendered correctly with reserved jafar field view props - `
    + `don't pass it to the underline component, pass the jafar props`, () => {
    element = shallow(<FieldView value="mouse" {...props} />);
    expect(element).toMatchSnapshot('with reserved prop');
  });
});
