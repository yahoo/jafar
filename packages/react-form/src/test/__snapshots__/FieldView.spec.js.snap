// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldView rendered correctly with disabled: with disabled field 1`] = `
<styled.div
  component="mockConstructor"
  id="mockId"
  margin="0 40px 40px 0"
>
  <styled.div>
    <styled.div
      color="#cccccc"
    >
      mockLabel
    </styled.div>
    <styled.div
      color="#cccccc"
    >
      *
    </styled.div>
    <styled.i
      title="mockDescription"
    >
      ?
    </styled.i>
  </styled.div>
  <div
    aria-label="Field Component"
  >
    <mockConstructor
      dirty={false}
      disabled={true}
      empty={false}
      invalid={false}
      onStateChange={[Function]}
      onValueChange={[Function]}
      required={true}
      state={Object {}}
      value="mockValue"
    />
  </div>
</styled.div>
`;

exports[`FieldView rendered correctly with invalid and required: invalid field required 1`] = `
<styled.div
  component="mockConstructor"
  id="mockId"
  margin="0 40px 40px 0"
>
  <styled.div>
    <styled.div
      color="#989697"
    >
      mockLabel
    </styled.div>
    <styled.div
      color="#989697"
    >
      *
    </styled.div>
    <styled.i
      title="mockDescription"
    >
      ?
    </styled.i>
  </styled.div>
  <div
    aria-label="Field Component"
  >
    <mockConstructor
      dirty={false}
      disabled={false}
      empty={false}
      invalid={false}
      onStateChange={[Function]}
      onValueChange={[Function]}
      required={true}
      state={Object {}}
      value="mockValue"
    />
  </div>
  <styled.div
    color="#3a3a3a"
  />
</styled.div>
`;

exports[`FieldView rendered correctly with invalid not required: invalid field not required 1`] = `
<styled.div
  component="mockConstructor"
  id="mockId"
  margin="0 40px 40px 0"
>
  <styled.div>
    <styled.div
      color="#e04336"
    >
      mockLabel
    </styled.div>
    <styled.div
      color="#e04336"
    >
      *
    </styled.div>
    <styled.i
      title="mockDescription"
    >
      ?
    </styled.i>
  </styled.div>
  <div
    aria-label="Field Component"
  >
    <mockConstructor
      dirty={false}
      disabled={false}
      empty={false}
      invalid={true}
      onStateChange={[Function]}
      onValueChange={[Function]}
      required={true}
      state={Object {}}
      value="mockValue"
    />
  </div>
  <styled.div
    color="#e04336"
  />
</styled.div>
`;

exports[`FieldView rendered correctly with reserved jafar field view props - don't pass it to the underline component, pass the jafar props: with reserved prop 1`] = `
<styled.div
  component="mockConstructor"
  id="mockId"
  margin="0 40px 40px 0"
>
  <styled.div>
    <styled.div
      color="#989697"
    >
      mockLabel
    </styled.div>
    <styled.div
      color="#989697"
    >
      *
    </styled.div>
    <styled.i
      title="mockDescription"
    >
      ?
    </styled.i>
  </styled.div>
  <div
    aria-label="Field Component"
  >
    <mockConstructor
      dirty={false}
      disabled={false}
      empty={false}
      invalid={false}
      onStateChange={[Function]}
      onValueChange={[Function]}
      required={true}
      state={Object {}}
      value="mockValue"
    />
  </div>
</styled.div>
`;

exports[`FieldView rendered correctly with unsupported jafar field props - pass it to the underline component: with unsupported prop 1`] = `
<styled.div
  component="mockConstructor"
  id="mockId"
  margin="0 40px 40px 0"
>
  <styled.div>
    <styled.div
      color="#989697"
    >
      mockLabel
    </styled.div>
    <styled.div
      color="#989697"
    >
      *
    </styled.div>
    <styled.i
      title="mockDescription"
    >
      ?
    </styled.i>
  </styled.div>
  <div
    aria-label="Field Component"
  >
    <mockConstructor
      dirty={false}
      disabled={false}
      empty={false}
      invalid={false}
      mickey="mouse"
      onStateChange={[Function]}
      onValueChange={[Function]}
      required={true}
      state={Object {}}
      value="mockValue"
    />
  </div>
</styled.div>
`;

exports[`FieldView rendered correctly: default values 1`] = `
<styled.div
  component="mockConstructor"
  id="mockId"
  margin="0 40px 40px 0"
>
  <styled.div>
    <styled.div
      color="#989697"
    >
      mockLabel
    </styled.div>
    <styled.div
      color="#989697"
    >
      *
    </styled.div>
    <styled.i
      title="mockDescription"
    >
      ?
    </styled.i>
  </styled.div>
  <div
    aria-label="Field Component"
  >
    <mockConstructor
      dirty={false}
      disabled={false}
      empty={false}
      invalid={false}
      onStateChange={[Function]}
      onValueChange={[Function]}
      required={true}
      state={Object {}}
      value="mockValue"
    />
  </div>
</styled.div>
`;
